client/src/js/services/Snapshot.js

```diff
@@ -0,0 +1,175 @@
+
+angular.module('bhima.services')
+.service('SnapshotService', SnapshotService);
+
+SnapshotService.$inject = ['$uibModal'];
+
+function SnapshotService($uibModal) {
+
+  var service = this;
+  service.dataUriToFile = dataUriToFile;
+  service.openWebcamModal = openWebcamModal;
+
+  function openWebcamModal() { 
+    return $uibModal.open({
+      templateUrl: 'modules/templates/bhSnapShot.html',
+      controller: snapshotController,
+      backdrop     : 'static',
+      animation    : false,
+      size : 'lg',
+    }).result;
+  }
+
+  // convert the data_url to a file object
+  function dataUriToFile(dataUri, fileName, mimeType) {
+    return (fetch(dataUri)
```

Comment:
I propose you use `$http` here instead of `fetch`.  The `fetch` API is relatively new in browsers.  In the same way that we use `$q` for promises until we can compile to any JS version, let's stick to what we know is supported.